### PR TITLE
Remember choice for default containers in the "Always open in" confirm page

### PR DESCRIPTION
**Pull request by rafeerahman**
**Before submitting your pull request**

- [x] I agree to license my code under the [MPL 2.0 license](https://www.mozilla.org/en-US/MPL/2.0/).
- [x] I rebased my work on top of the main branch.
- [x] I ran `npm test` and all tests passed.
- [x] I added test coverages if relevant.

# Description

When a user is on the confirm-page after re-opening a site they chose to 'always open in a container', if they click "Remember my choice" and then "Open in current tab" (default container), this is not remembered. i.e. The next time they open the site, they are prompted to this screen again. 

This PR fixes that. 

image

## How to test

(Remember choice for default container)
   * Go to youtube.com
   * In the browserAction menu, choose "Always Open This Site in..."
   *  Choose the work container
   * Open youtube.com in a new tab.
   * On the confirmation page prompt, check "Remember my decision for this site" then click "Open in current tab"
   * Open youtube.com in a new tab, and verify that there is no confirmation page. (if you want to put it in a container again, click "Always Open This Site In....")
   * In the work containers "site list" verify that it is empty.

(Switching between two containers)
  * Go to youtube.com
  * In the browserAction menu, choose "Always Open This Site in..."
  * Choose the Work container, youtube will open in this container
  * In the browserAction menu, choose "Always Open This Site in..."
  * Choose the Personal container
  * On the confirmation page, check "remember my decision for this site".
  * Choose the "Work" container (the option on the left).
  * Verify that the choice is remembered.

## Type of change

*Select all that apply.*

- [x] Bug fix
- [ ] New feature
- [ ] Major change (fix or feature that would cause existing functionality to work differently than in the current version)

Tag issues related to this pull request:

* https://github.com/mozilla/multi-account-containers/issues/2603

### CODE REVIEW

**Comment by rafeerahman** [comment]
Also looking into integrating / fixing https://github.com/mozilla/multi-account-containers/pull/2617, which tries to remember the choice when switching to a container that is not the "default container" (i.e. not "Open in current tab").

**Review by groovecoder — CHANGES_REQUESTED** [review]
Couple blocking changes.

Can you write out the steps you used to spot-check this? I tried:

1. Go to accounts.firefox.com
2. In the browserAction menu, choose "Always Open This Site in..."
3. In the list of containers, choose "Work"
4. Close the tab
5. Open a new window
6. Go to accounts.firefox.com

Expected Result:
* I should be prompted to confirm I want to open the site in a Work tab

Actual Result:
* I'm not prompted to confirm I want to open the site in a Work tab

**Comment by groovecoder** [comment]
Also, please change the commit message to start with `fix #2603:` so that GitHub will automatically link this PR with the corresponding issue.

File: src/js/background/assignManager.js [hunk 1]
```diff
@@ -764,7 +771,7 @@ window.assignManager = {
   async initBookmarksMenu() {
     browser.contextMenus.create({
       id: this.OPEN_IN_CONTAINER,
-      title: "Open Bookmark in Container Tab",
+      title: browser.i18n.getMessage("openBookmarkInContainerTab"),
```
[the remark this answers is not in the document]
Comment (rafeerahman) [reply src/js/background/assignManager.js]: looks like I amended to most recent merge commit by accident

**Review by groovecoder — CHANGES_REQUESTED** [review]
Thanks @rafeerahman I was able to reproduce the bug and the fix on youtube.com.

There's still one more bug that I think is worth fixing:

1. Go to youtube.com
2. In the browserAction menu, choose "Always Open This Site in..."
3. Choose the work container
4. Open youtube.com in a new tab.
5. On the confirmation page prompt, check "Remember my decision for this site" then click "Open in current tab"
6. Open youtube.com in a new tab
   * [ ] verify that there is no confirmation page.
   * [ ] verify youtube.com opens in default/non-container tab
8. Open the work containers "site list"
   * [ ] verify that it is empty.
9. Open a new work tab
10. Go to youtube.com

Expected results:
youtube.com automatically opens in default/non-container tab without prompting

Actual results:
youtube.com opens in work container tab

File: src/js/confirm-page.js [hunk 2]
```diff
@@ -72,8 +76,30 @@ function getCurrentTab() {
...
+  }
+
+  if (neverAsk) {
+    browser.runtime.sendMessage({
+      method: "neverAsk",
+      neverAsk: true,
+      cookieStoreId: currentCookieStoreId,
+      pageUrl: redirectUrl
+    });
+  }
```
Comment (achernyakevich-sc) [inline src/js/confirm-page.js:101]: Maybe optimize it by something like:

```
if (neverAsk) {
  await browser.runtime.sendMessage({
    method: "neverAsk",
    neverAsk: true,
    cookieStoreId: currentCookieStoreId,
    pageUrl: redirectUrl,
    defaultContainer: !currentContainer
  });
}
```

Comment (rafeerahman) [reply src/js/confirm-page.js:101]: Thanks!

**Review by groovecoder — APPROVED** [review]

**Pull request merged**